test/blackbox-tests/test-cases/github5555.t

```diff
@@ -0,0 +1,6 @@
+  $ cat > dune-workspace << EOF
+  > (lang dune 2.9)
+  > (env (_ (binaries x.exe)))
+  > EOF
+
+  $ dune build
```

Comment:
Could you please:

* Add a description to this test:
* Split the test into a separate commit that demonstrates the failure.